servicetalk-opentelemetry-http/src/main/java/io/servicetalk/opentelemetry/http/ScopeTracker.java

```diff
@@ -22,54 +22,100 @@
 import io.servicetalk.http.api.HttpResponseMetaData;
 import io.servicetalk.http.api.StreamingHttpRequest;
 import io.servicetalk.http.api.StreamingHttpResponse;
-import io.servicetalk.http.utils.BeforeFinallyHttpOperator;
+import io.servicetalk.http.utils.AfterFinallyHttpOperator;
 
 import io.opentelemetry.context.Context;
 import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
 
+import java.util.concurrent.atomic.AtomicIntegerFieldUpdater;
 import javax.annotation.Nullable;
 
 import static java.util.Objects.requireNonNull;
 
-class ScopeTracker implements TerminalSignalConsumer {
+final class ScopeTracker implements TerminalSignalConsumer {
```

Comment:
Definitely not in this PR, but maybe later we can offer tracker as a public utility for other use-cases